File: Backend/KebunJio/src/main/java/iss/nus/edu/sg/sa4106/KebunJio/Models/Reminder.java
package iss.nus.edu.sg.sa4106.KebunJio.Models;

import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.annotation.JsonFormat;

@Document(collection = "Reminders")
public class Reminder {
	@Id
	private String id;  
    private String userId;
    private String plantId; 
    private String reminderType;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime reminderDateTime;
    private Boolean isRecurring;
    private String recurrenceInterval;
    private String status;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdDateTime; 

    
    public Reminder() {}


	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getUserId() {
		return userId;
	}

	public void setUserId(String userId) {
		this.userId = userId;
	}

	public String getPlantId() {
		return plantId;
	}

	public void setPlantId(String plantId) {
		this.plantId = plantId;
	}

	public String getReminderType() {
		return reminderType;
	}

	public void setReminderType(String reminderType) {
		this.reminderType = reminderType;
	}

	public LocalDateTime getReminderDateTime() {
		return reminderDateTime;
	}

	public void setReminderDateTime(LocalDateTime reminderDateTime) {
		this.reminderDateTime = reminderDateTime;
	}

	public Boolean getIsRecurring() {
		return isRecurring;
	}

	public void setIsRecurring(Boolean isRecurring) {
		this.isRecurring = isRecurring;
	}

	public String getRecurrenceInterval() {
		return recurrenceInterval;
	}

	public void setRecurrenceInterval(String recurrenceInterval) {
		this.recurrenceInterval = recurrenceInterval;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public LocalDateTime getCreatedDateTime() {
		return createdDateTime;
	}

	public void setCreatedDateTime(LocalDateTime createdDateTime) {
		this.createdDateTime = createdDateTime;
	} 

}

File: Mobile/app/src/main/java/iss/nus/edu/sg/sa4106/kebunjio/data/Reminder.kt
package iss.nus.edu.sg.sa4106.kebunjio.data

import iss.nus.edu.sg.sa4106.kebunjio.HandleNulls.Companion.ifNullBoolean
import java.io.Serializable
import iss.nus.edu.sg.sa4106.kebunjio.HandleNulls.Companion.ifNullString
import org.json.JSONObject
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

data class Reminder(
    val id: String,
    val userId: String,
    val plantId: String,
    val reminderType: String,
    val reminderDateTime: String,
    val isRecurring: Boolean,
    val recurrenceInterval: String,
    val status: String,
    val createdDateTime: LocalDateTime
) : Serializable {
    companion object {
        fun getFromResponseObject(responseObject: JSONObject): Reminder {
            val id = ifNullString(responseObject.getString("id"))
            val userId = ifNullString(responseObject.getString("userId"))
            val plantId = ifNullString(responseObject.getString("plantId"))
            val reminderType = ifNullString(responseObject.getString("reminderType"))
            val reminderDateTime = ifNullString(responseObject.getString("reminderDateTime"))
            val isRecurring = ifNullBoolean(responseObject.getBoolean("isRecurring"))
            val recurrenceInterval = ifNullString(responseObject.getString("recurrenceInterval"))
            val status = ifNullString(responseObject.getString("status"))
            val createdDateTimeString = ifNullString(responseObject.getString("createdDateTime"))

            val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss")
            val createdDateTime = LocalDateTime.parse(createdDateTimeString, formatter)

            return Reminder(id, userId, plantId, reminderType, reminderDateTime, isRecurring, recurrenceInterval, status, createdDateTime)
        }
    }
}
